src/main/java/org/folio/fqm/service/FqlToSqlConverterService.java

```diff
@@ -223,22 +227,26 @@ private static Condition handleRegEx(RegexCondition regexCondition, org.jooq.Fie
     return condition("{0} ~* {1}", field, val(regexCondition.value()));
   }
 
-  private static Condition handleContains(ContainsCondition containsCondition, EntityType entityType, org.jooq.Field<Object> field) {
-    if (containsCondition.value() instanceof String value) {
-      // Casting required to use ARRAY_CONTAINS operator
-      return cast(field, String[].class).contains(valueField(new String[]{value.toLowerCase()}, containsCondition, entityType));
-    }
-    // Cast non-string types to string for easy comparison
-    return cast(field, String[].class).contains(valueField(new String[]{containsCondition.value().toString()}, containsCondition, entityType));
+  private static Condition handleContainsAll(ContainsAllCondition containsAllCondition, EntityType entityType, org.jooq.Field<Object> field) {
+    var valueList = containsAllCondition
+      .value()
+      .stream()
+      .map(val -> valueField(val, containsAllCondition, entityType))
+      .toList()
+      .toArray(new org.jooq.Field[0]);
```

Comment:
Rather than collect into a list and convert that into an array, you can go straight into an array with
```suggestion
      .toArray(org.jooq.Field[]::new);
```